AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSGraphQLSubscriptionOperation.swift

```diff
@@ -64,58 +64,125 @@ final public class AWSGraphQLSubscriptionOperation<R: Decodable>: GraphQLSubscri
+                                                                     eventHandler: { [weak self] event, _ in
+                    self?.onAsyncSubscriptionEvent(event: event)
+                })
+            }
         }
+    }
 
-        // Retrieve endpoint configuration
-        let endpointConfig: AWSAPICategoryPluginConfiguration.EndpointConfig
+    private func validate<T>(request: GraphQLOperationRequest<T>) -> Result<Void, APIError> {
         do {
-            endpointConfig = try pluginConfig.endpoints.getConfig(for: request.apiName, endpointType: .graphQL)
+            try request.validate()
+            return .success(())
         } catch let error as APIError {
-            dispatch(result: .failure(error))
-            finish()
-            return
+            return .failure(error)
         } catch {
-            dispatch(result: .failure(APIError.unknown("Could not get endpoint configuration", "", nil)))
-            finish()
-            return
+            return .failure(APIError.unknown("Could not validate request", "", nil))
         }
+    }
 
-        // Retrieve request plugin option and
-        // auth type in case of a multi-auth setup
-        let pluginOptions = request.options.pluginOptions as? AWSPluginOptions
+    private func getEndpointConfig(
+        from pluginConfig: AWSAPICategoryPluginConfiguration,
+        apiName: String?
+    ) -> Result<AWSAPICategoryPluginConfiguration.EndpointConfig, APIError> {
+        do {
+            return .success(try pluginConfig.endpoints.getConfig(for: apiName, endpointType: .graphQL))
+        } catch let error as APIError {
+            return .failure(error)
+        } catch {
+            return .failure(APIError.unknown("Could not get endpoint configuration", "", nil))
+        }
+    }
 
-        // Retrieve the subscription connection
-        subscriptionQueue.sync {
-            do {
-                subscriptionConnection = try subscriptionConnectionFactory
-                    .getOrCreateConnection(for: endpointConfig,
-                                              authService: authService,
-                                              authType: pluginOptions?.authType,
-                                              apiAuthProviderFactory: apiAuthProviderFactory)
-            } catch {
-                let error = APIError.operationError("Unable to get connection for api \(endpointConfig.name)", "", error)
-                dispatch(result: .failure(error))
-                finish()
-                return
-            }
+    private func makeURLRequest(
+        with endpointConfig: AWSAPICategoryPluginConfiguration.EndpointConfig
+    ) -> URLRequest {
+        var urlRequest = URLRequest(url: endpointConfig.baseURL)
+        urlRequest.setValue(AmplifyAWSServiceConfiguration.baseUserAgent(),
+                            forHTTPHeaderField: URLRequestConstants.Header.userAgent)
```

Comment:
we should cherry-pick this change and release it as is, without enabling subscription interceptors. This PR can continue to be the proof of concept for enabling that use case, but requires some cross platform involvement to make sure we have a consolidated developer experience